vllm/engine/llm_engine.py

```diff
@@ -575,26 +577,51 @@ def _get_stats(self,
         # Iteration stats if we have scheduler output.
         num_prompt_tokens = 0
         num_generation_tokens = 0
+        num_prompt_tokens_lst = []
+        num_generation_tokens_lst = []
+        request_n = []
+        request_best_of = []
         time_to_first_tokens = []
         time_per_output_tokens = []
         time_e2e_requests = []
+        finished_reason_counter = CollectionsCounter()
         if scheduler_outputs is not None:
             prompt_run = scheduler_outputs.num_prefill_groups > 0
 
-            # Number of Tokens.
+            # Number of Tokens
             if prompt_run:
-                num_prompt_tokens = sum(
+                num_prompt_tokens_lst = [
```

Comment:
This is actually an incorrect calculation. 

This `stat` is used to compute the histogram of prompt lengths for each request. Because of chunked prefill, the prompt processing will occur over multiple `steps` of LLMEngine. So this will actually give you the distribution of chunks as opposed to the distribution of prefill lengths.

So what I would suggest is that we log the of the generation and the prompt once the `seq_group` is finished (e.g. in the loop where we loop through the `scheduled_sequence_groups`

This way we know that we are logging the full prefill length and that we are logging one item per iteration